portfolio/src/main/java/com/google/sps/servlets/TitanicDataServlet.java

```diff
@@ -0,0 +1,72 @@
+
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.LinkedHashMap;
+import java.util.Scanner;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Returns titanic survivor data by class in a hash map*/
+@WebServlet("/titanic-data")
+public class TitanicDataServlet extends HttpServlet {
+
+  private LinkedHashMap<String, Double> titanicSurvivors = new LinkedHashMap<>();
+
+  @Override
+  public void init() {
+    Double firstPercent = 0.0;
+    Double secondPercent = 0.0;
+    Double thirdPercent = 0.0;
+
+    Double firstTotal = 0.0;
+    Double secondTotal = 0.0;
+    Double thirdTotal = 0.0;
+
+    Double firstSurvived = 0.0;
+    Double secondSurvived = 0.0;
+    Double thirdSurvived = 0.0;
+
+    Scanner scanner = new Scanner(getServletContext().getResourceAsStream(
+        "/WEB-INF/train.csv"));
+    while (scanner.hasNextLine()) {
+      String line = scanner.nextLine();
+      String[] cells = line.split(",");
+
+      Integer survived = Integer.valueOf(cells[0]);
+      Integer shipClass = Integer.valueOf(cells[1]);
```

Comment:
make private static final constants for these cells.  This will make it easier to change in the future if the file format changes and also benefits from more explicit documentation in the code.

Reply:
@megankj Just to clarify, do you mean making "cells[0]" and "cells[1]" into constants?

Reply:
The numbers 0 and 1